### PR TITLE
[TK-9]: otp screen ui

### PR DESCRIPTION
**Pull request by AbdulsKhan1**
## Jira: [TK-9](https://shop-go.atlassian.net/browse/TK-9?atlOrigin=eyJpIjoiMDI4OTMyZmY3YjVjNGQyNTliZGEyOTEyNjI0NmNiY2IiLCJwIjoiaiJ9)

**Description**:
- add 4 textfield with functionality of auto moving to next textfield when filled.
- put an image, title label and continue button with a corner radius of 22.
- add auto layout.

**How to test it?**
- type in textfield.
- click continue button.
- try in different simulator.

### CODE REVIEW

**Review by basitalich — CHANGES_REQUESTED** [review]

File: Vape-X/OTPScreenViewController.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            case OTPSecondTextField:
+                OTPThirdTextField.becomeFirstResponder()
+            case OTPThirdTextField:
+                OTPFourthTextField.becomeFirstResponder()
+            case OTPFourthTextField:
+                OTPFourthTextField.resignFirstResponder()
+            default:
+                break
+            }
+        }else if text?.count == 0 {
```
Comment (haroonmaqsoodch) [inline Vape-X/OTPScreenViewController.swift:51]: space before else

File: Vape-X/OTPScreenViewController.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                OTPSecondTextField.becomeFirstResponder()
+            case OTPSecondTextField:
+                OTPThirdTextField.becomeFirstResponder()
+            case OTPThirdTextField:
+                OTPFourthTextField.becomeFirstResponder()
+            case OTPFourthTextField:
+                OTPFourthTextField.resignFirstResponder()
+            default:
+                break
+            }
```
Comment (haroonmaqsoodch) [inline Vape-X/OTPScreenViewController.swift:50]: one space after curly brace

File: Vape-X/OTPScreenViewController.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        OTPFirstTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        OTPSecondTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        OTPThirdTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        OTPFourthTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        createAccountOTPButton.layer.cornerRadius = 35
+        createAccountOTPButton.layer.masksToBounds = true
+        
+        //hello
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift]: fixed.

File: Vape-X/OTPScreenViewController.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    
+    @objc func textDidChange(textField: UITextField) {
+        let text = textField.text
+        if text?.count == 1{
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift]: fixed.

File: Vape-X/OTPScreenViewController.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                OTPThirdTextField.becomeFirstResponder()
+            case OTPThirdTextField:
+                OTPFourthTextField.becomeFirstResponder()
+            case OTPFourthTextField:
+                OTPFourthTextField.resignFirstResponder()
+            default:
+                break
+            }
+        }else if text?.count == 0{
+            switch textField{
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift]: fixed.

File: Vape-X/OTPScreenViewController.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        createAccountOTPButton.layer.masksToBounds = true
+        
+        //hello
+    }
+    
+    @objc func textDidChange(textField: UITextField) {
+        let text = textField.text
+        if text?.count == 1{
+            switch textField{
+            case OTPFirstTextField:
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift:40]: fixed.

File: Vape-X/OTPScreenViewController.swift [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    @IBOutlet weak var createAccountOTPButton: UIButton!
+    @IBOutlet weak var OTPFirstTextField: UITextField!
+    @IBOutlet weak var OTPSecondTextField: UITextField!
+    @IBOutlet weak var OTPThirdTextField: UITextField!
+    @IBOutlet weak var OTPFourthTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        OTPFirstTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift:30]: fixed.

File: Vape-X/OTPScreenViewController.swift (hunk 1 above)
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift:51]: fixed

File: Vape-X/OTPScreenViewController.swift (hunk 2 above)
Comment (AbdulsKhan1) [reply Vape-X/OTPScreenViewController.swift:50]: fixed.

File: Vape-X/Base.lproj/Main.storyboard [hunk 8]
```diff
@@ -1,24 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="13122.16" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="BYZ-38-t0r">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="21225" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="BYZ-38-t0r">
+    <device id="retina6_0" orientation="portrait" appearance="light"/>
     <dependencies>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="13104.12"/>
```
[the remark this answers is not in the document]
Comment (AbdulsKhan1) [reply Vape-X/Base.lproj/Main.storyboard:4]: this change was made after the recent pull this is not my change.

**Review by ammadahmed22 — APPROVED** [review]
LGTM

File: Vape-X/Base.lproj/Main.storyboard (hunk 8 above)
[the remark this answers is not in the document]
Comment (Annas90) [reply Vape-X/Base.lproj/Main.storyboard:4]: where is the vapes picture

**Pull request merged**